tests/test_abacus_pw_scf.py

```diff
@@ -88,15 +89,15 @@ def test_force(self) :
         # fp.close()
 
     def test_virial(self) :
-        fp = open('abacus.scf/ch4_virial')
-        virial = []
-        for ii in fp :
-            virial.append([float(jj) for jj in ii.split()])
-        virial = np.array(virial)
-        for ii in range(virial.shape[0]) :
-            for jj in range(virial.shape[1]) :
-                self.assertAlmostEqual(self.system_ch4.data['virials'][0][ii][jj], virial[ii][jj], places = 3)
-        fp.close()
+        with open('abacus.scf/ch4_virial') as fp:
+            virial = []
+            for ii in fp :
+                virial.append([float(jj) for jj in ii.split()])
+            virial = np.array(virial)
+            for ii in range(virial.shape[0]) :
+                for jj in range(virial.shape[1]) :
+                    self.assertAlmostEqual(self.system_ch4.data['virials'][0][ii][jj], virial[ii][jj], places = 3)
```

Comment:
`np.testing.assert_almost_equal` can be used to compare numpy arrays.

Reply:
Thank you for the most helpful suggestion! I have changed all possible cases in both `test_abacus_pw_scf.py` and `test_abacus_md.py` into `np.testing.assert_almost_equal`.